pxtblocks/blocklyloader.ts

```diff
@@ -2635,6 +2636,29 @@ namespace pxt.blocks {
                 setOutputCheck(this, typeName, cachedBlockInfo);
             };
         }
+
+        /**
+         * Make a context menu option for creating a function call block.
+         * This appears in the context menu for function definitions.
+         * @param {!Blockly.BlockSvg} block The block where the right-click originated.
+         * @return {!Object} A menu option, containing text, enabled, and a callback.
+         * @package
+         */
+        (Blockly as any).Functions.makeCreateCallOption = function (block: Blockly.Block) {
+            let functionName = block.getField("function_name").getText();
+
+            let mutation = goog.dom.createDom('mutation');
+            mutation.setAttribute('name', functionName);
+            let callBlock = goog.dom.createDom('block', null, mutation);
+            callBlock.setAttribute('type', 'function_call');
+
+            let option = {
+                enabled: block.workspace.remainingCapacity() > 0,
+                text: Util.lf("Create 'call {0}'", functionName),
```

Comment:
Is the only functionality that needs to be patched the `text` property? Would it be better to avoid the code duplication by calling the original method and then overwrite the `text` property of the return value?

```ts
const makeCreateCallOptionOriginal = (Blockly as any).Functions.makeCreateCallOption;
(Blockly as any).Functions.makeCreateCallOption = function (block: Blockly.Block) {
   let option = makeCreateCallOptionOriginal(block);
   let functionName = block.getField("function_name").getText();
   option.text = Util.lf("Create 'call {0}'", functionName);
   return option;
}
```

Reply:
true, I'll make the update!